xdp-loader/xdp-loader.c

```diff
@@ -271,31 +287,43 @@ int print_iface_status(const struct iface *iface,
 	}
 
 	dispatcher = xdp_multiprog__main_prog(mp);
+	if (dispatcher) {
+		printf("%-16s %-5s %-16s %-8s %-4d %-17s\n",
+		iface->ifname,
+		"",
+		xdp_program__name(dispatcher),
+		get_enum_name(xdp_modes, xdp_multiprog__attach_mode(mp)),
+		xdp_program__id(dispatcher),
+		print_bpf_tag(tag, xdp_program__tag(dispatcher)));
+
+
+		for (prog = xdp_multiprog__next_prog(NULL, mp);
+		     prog;
+		     prog = xdp_multiprog__next_prog(prog, mp)) {
+
+			err = xdp_program__print_chain_call_actions(prog, buf,
+								    sizeof(buf));
+			if (err)
+				return err;
+
+			printf("%-16s %-5d %-16s %-8s %-4u %-17s %s\n",
+			       " =>", xdp_program__run_prio(prog),
+			       xdp_program__name(prog),
+			       "", xdp_program__id(prog),
+			       print_bpf_tag(tag, xdp_program__tag(prog)),
+			       buf);
+		}
+	}
 
-	printf("%-16s %-5s %-16s %-8s %-4d %-17s\n",
-	       iface->ifname,
-	       "",
-	       xdp_program__name(dispatcher),
-	       get_enum_name(xdp_modes, xdp_multiprog__attach_mode(mp)),
-	       xdp_program__id(dispatcher),
-	       print_bpf_tag(tag, xdp_program__tag(dispatcher)));
-
-
-	for (prog = xdp_multiprog__next_prog(NULL, mp);
-	     prog;
-	     prog = xdp_multiprog__next_prog(prog, mp)) {
-
-		err = xdp_program__print_chain_call_actions(prog, buf,
-							    sizeof(buf));
-		if (err)
-			return err;
-
-		printf("%-16s %-5d %-16s %-8s %-4u %-17s %s\n",
-		       " =>", xdp_program__run_prio(prog),
-		       xdp_program__name(prog),
-		       "", xdp_program__id(prog),
-		       print_bpf_tag(tag, xdp_program__tag(prog)),
-		       buf);
+	hw_prog = xdp_multiprog__hw_prog(mp);
```

Comment:
I think we should print this before the others, as HW progs are executed first...